tests/org.jboss.tools.openshift.ui.bot.test/src/org/jboss/tools/openshift/ui/bot/test/application/v3/adapter/ImportApplicationWizardTest.java

```diff
@@ -0,0 +1,161 @@
+/*******************************************************************************
+ * Copyright (c) 2017 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v 1.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributor:
+ *     Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+package org.jboss.tools.openshift.ui.bot.test.application.v3.adapter;
+
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+
+import org.eclipse.jface.wizard.WizardDialog;
+import org.eclipse.swt.widgets.Shell;
+import org.jboss.reddeer.common.wait.TimePeriod;
+import org.jboss.reddeer.common.wait.WaitUntil;
+import org.jboss.reddeer.common.wait.WaitWhile;
+import org.jboss.reddeer.core.util.Display;
+import org.jboss.reddeer.eclipse.jdt.ui.ProjectExplorer;
+import org.jboss.reddeer.swt.impl.shell.DefaultShell;
+import org.jboss.tools.openshift.reddeer.condition.AmountOfResourcesExists;
+import org.jboss.tools.openshift.reddeer.condition.OpenShiftResourceExists;
+import org.jboss.tools.openshift.reddeer.enums.Resource;
+import org.jboss.tools.openshift.reddeer.enums.ResourceState;
+import org.jboss.tools.openshift.reddeer.requirement.OpenShiftCommandLineToolsRequirement.OCBinary;
+import org.jboss.tools.openshift.reddeer.requirement.OpenShiftConnectionRequirement.RequiredBasicConnection;
+import org.jboss.tools.openshift.reddeer.utils.OpenShiftLabel;
+import org.jboss.tools.openshift.reddeer.utils.TestUtils;
+import org.jboss.tools.openshift.reddeer.view.OpenShiftExplorerView;
+import org.jboss.tools.openshift.reddeer.view.resources.Service;
+import org.jboss.tools.openshift.reddeer.wizard.importapp.GitCloningWizardPage;
+import org.jboss.tools.openshift.reddeer.wizard.importapp.ImportApplicationWizard;
+import org.jboss.tools.openshift.reddeer.wizard.server.ServerSettingsWizard;
+import org.jboss.tools.openshift.reddeer.wizard.server.ServerSettingsWizardPage;
+import org.jboss.tools.openshift.ui.bot.test.application.v3.create.AbstractCreateApplicationTest;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+/**
+ * @author jnovak@redhat.com
+ */
+@OCBinary
+@RequiredBasicConnection
+public class ImportApplicationWizardTest extends AbstractCreateApplicationTest {
+
+	@BeforeClass
+	public static void waitTillApplicationIsRunning() {
+		new WaitWhile(new OpenShiftResourceExists(Resource.BUILD, "eap-app-1", ResourceState.RUNNING),
+				TimePeriod.getCustom(600));
+		new WaitUntil(new AmountOfResourcesExists(Resource.POD, 2), TimePeriod.LONG, false);
```

Comment:
I rewrote it with service requirement. I also had to edit work with OC project, because annotation params must be constants. (So far I used project with name generated in runtime - DatastoreOS3.PROJECT1)